specs/tendrl_performance_enhacements.adoc

```diff
@@ -0,0 +1,235 @@
+
+```
+counter = 1
+while True:
+    sleep(10)
+    sync volume and bricks
+
+    if counter % 30 == 0: # if 30 rounds of volume sync has happened, trigger
+        sync cluster status
+        sync utilization details
+
+    if counter % 60 == 0: # if 60 rounds of volume sync has happened, trigger
+        sync snapshots
+        sync underlying device details for bricks
+
+    counter = (counter + 1) % 60 (LCM of 1, 30, 60)
+```
+
+This would make sure different syncs are done at different intervals. Also this
+does not require sds_sync to be a separate flow and still different syncs get
+triggered at different intervals.
+
+=== Data model impact
+
+* Annotate the tendrl flows in different definitions files of tendrl modules to
+define the tagged queue name where these jobs would be written
+
+=== Impacted Modules:
+
+==== Tendrl API impact:
+
+* With proposed changes above, the object details would be saved as single JSON
+field with name `data`. For example, the volume details would be saved as
+`clusters/{int-id}/Volumes/{vol-id}/data`. API layer need to change to read the
+values as per these changes for listing the objects details.
+
+* REST layer to write the jobs in tagged queues based on definitions
+
+* Enhancements for tuning the response time for various GET endpoints
+
+==== Notifications/Monitoring impact:
+None
+
+==== Tendrl/common impact:
+
+* Enhancements for processing tagged job queues. Based on the current service,
+it should look at defined tagged job queue only for figuring out the jobs to be
+picked and processed
+
+* Enhance the writing/reading logic to/from etcd to consider the whole object as
+single JSON
+
+==== Tendrl/node_agent impact:
+
+* Definitions changes for tagging flows with specific job queue names
+
+==== Sds integration impact:
+
+* Definitions changes for tagging flows with specific job queue names
+
+==== Tendrl Dashboard impact:
+
+None
+
+=== Security impact:
+
+None.
+
+=== Other end user impact:
+
+None
+
+=== Performance impact:
+
+None.
+
+=== Other deployer impact:
+
+None.
+
+=== Developer impact:
+
+None.
+
+
+== Implementation:
+
+* https://github.com/Tendrl/documentation/issues/88
+
+* https://github.com/Tendrl/documentation/issues/89
+
+* https://github.com/Tendrl/documentation/issues/90
+
+* https://github.com/Tendrl/commons/issues/657
+
+=== Assignee(s):
+
+Primary assignee:
+  shtripat
+  r0h4n
+  anivargi
+
+=== Work Items:
+
+* https://github.com/Tendrl/specifications/issues/172
+
+
+== Dependencies:
+
+None
+
+
+== Testing:
+
+* Verify that load incurred on storage nodes due to tendrl components is within
+the defined limits
+
+* Verify the REST endpoints for their response time and it should be within the
+defined time limits
+
+* Verify the guidelines published regarding clustering of etcd
+
+* Verify all the objects listing REST endpoints to make sure all the details are
+listed properly.
+
+
+== Documentation impact:
+
+* Document for clustered setup of etcd
```

Comment:
After discussion with dev team, this was moved out into [3rd milestone](https://github.com/Tendrl/specifications/milestone/1).